src/client.ts

```diff
@@ -1511,27 +1511,99 @@ export class StreamChat<
-        EventType,
-        MessageType,
-        ReactionType,
-        UserType
-      >(this, channelType, undefined, channelIDOrCustom);
+      return this.getChannelByMembers(channelType, channelIDOrCustom);
     }
 
-    if (typeof channelIDOrCustom === 'string' && ~channelIDOrCustom.indexOf(':')) {
-      throw Error(
-        `Invalid channel id ${channelIDOrCustom}, can't contain the : character`,
-      );
+    return this.getChannelById(channelType, channelIDOrCustom, custom);
+  }
+
+  /**
+   * Its a submethod for `client.channel()` method, used to create unique conversation or
+   * channel based on member list instead of id.
+   *
+   * If the channel already exist in `activeChannels` list, then we simply return it, since that
+   * means the same channel was already requested or created.
+   *
+   * Otherwise we create new instance of Channel class and return it.
+   *
+   * @param {string} channelType The channel type
+   * @param {object} [custom]    Custom data to attach to the channel
+   *
+   * @return {channel} The channel object, initialize it using channel.watch()
+   */
+  getChannelByMembers = (channelType: string, custom: ChannelData<ChannelType>) => {
+    // Check if the channel already exists.
+    // Only allow 1 channel object per cid
+    const membersStr = custom.members?.sort().join(',');
+    if (!membersStr) {
+      throw Error('Please specify atleast one member when creating unique conversation');
+    }
+
+    // channel could exist in `activeChannels` list with either one of the following two keys:
+    // 1. cid - Which gets set on channel only after calling channel.query or channel.watch or channel.create
+    // 2. Sorted membersStr - E.g., "messaging:amin,vishal" OR "messaging:amin,jaap,tom"
+    //                        This gets set when you create a channel, but haven't queried yet. After query,
+    //                        we will replace it with `cid`
+    for (const key in this.activeChannels) {
+      const channel = this.activeChannels[key];
+      if (key === `${channelType}:${membersStr}`) {
+        return channel;
+      }
+
+      if (key.indexOf(`${channelType}:!members-`) === 0) {
+        const membersStrInExistingChannel = Object.keys(channel.state.members)
+          .sort()
+          .join(',');
+        if (membersStrInExistingChannel === membersStr) {
+          return channel;
+        }
+      }
+    }
+
+    const channel = new Channel<
+      AttachmentType,
+      ChannelType,
+      CommandType,
+      EventType,
+      MessageType,
+      ReactionType,
+      UserType
+    >(this, channelType, undefined, custom);
+
+    // For the time being set the key as membersStr, since we don't know the cid yet.
+    // In channel.query, we will replace it with 'cid'.
+    this.activeChannels[`${channelType}:${membersStr}`] = channel;
+    return channel;
+  };
+
+  /**
+   * Its a submethod for `client.channel()` method, used to channel given the id of channel.
```

Comment:
It's

Reply:
`submethod` doesn't intuitive, `a helper` maybe

Reply:
also, let's add a `@private` to these two functions

Reply:
Done :)